subprojects/samples/src/integTest/groovy/org/gradle/integtests/samples/files/SamplesCopyIntegrationTest.groovy

```diff
@@ -240,6 +241,141 @@ class SamplesCopyIntegrationTest extends AbstractSampleIntegrationTest {
         dsl << ['groovy', 'kotlin']
     }
 
+    @UsesSample("files/copy")
+    def "can use copy task with #dsl dsl"() {
+        given:
+        def dslDir = sample.dir.file(dsl)
+        executer.inDirectory(dslDir)
+
+        when:
+        succeeds('copyTask')
+
+        then:
+        def outputDir = dslDir.file("build/explodedWar")
+        outputDir.file('web.xml').isFile()
+        outputDir.file('index-staging.html').isFile()
+        outputDir.file('products/gradle-staging.html').isFile()
+
+        where:
+        dsl << ['groovy', 'kotlin']
+    }
+
+    @UsesSample("files/copy")
+    def "can use copy task with patterns with #dsl dsl"() {
+        given:
+        def dslDir = sample.dir.file(dsl)
+        executer.inDirectory(dslDir)
+
+        when:
+        succeeds('copyTaskWithPatterns')
+
+        then:
+        def outputDir = dslDir.file("build/explodedWar")
+        outputDir.file('web.xml').assertDoesNotExist()
+        outputDir.file('index-staging.html').isFile()
+        outputDir.file('products/gradle-staging.html').isFile()
+        outputDir.file('draft.html').assertDoesNotExist()
+
+        where:
+        dsl << ['groovy', 'kotlin']
+    }
+
+    @UsesSample("files/copy")
+    def "can use copy task with multiple from clauses with #dsl dsl"() {
+        given:
+        def dslDir = sample.dir.file(dsl)
+        executer.inDirectory(dslDir)
+
+        when:
+        succeeds('anotherCopyTask')
+
+        then:
+        def outputDir = dslDir.file("some-dir")
+        outputDir.file('web.xml').isFile()
+        outputDir.file('index-staging.html').isFile()
+        outputDir.file('products/gradle-staging.html').isFile()
+        outputDir.file('asset1.txt').isFile()
+
+        where:
+        dsl << ['groovy', 'kotlin']
+    }
+
+    @UsesSample("files/copy")
+    @ToBeFixedForConfigurationCache(because = "https://github.com/gradle/gradle/issues/22536")
```

Comment:
My plan is to add a warning similar to what we're going to have in #22683 to the doc and rewrite the snippet as part of #22536.